asdf_astropy/tags/transform/basic.py

```diff
@@ -0,0 +1,216 @@
+
+
+class TransformType(asdf.AsdfConverter):
+    tags = {"http://asdf-format.org/schemas/transform/transforms-1.0.0"}
+    types = {Model}
+    version = '2.0.0'
+    requires = ['astropy']
+
+    def _from_tree_base_transform_members(self, model, node):
+        if 'name' in node:
+            model.name = node['name']
+
+        if 'bounding_box' in node:
+            model.bounding_box = node['bounding_box']
+
+        if "inputs" in node:
+            if model.n_inputs == 1:
+                model.inputs = (node["inputs"],)
+            else:
+                model.inputs = tuple(node["inputs"])
+
+        if "outputs" in node:
+            if model.n_outputs == 1:
+                model.outputs = (node["outputs"],)
+            else:
+                model.outputs = tuple(node["outputs"])
+
+        param_and_model_constraints = {}
+        for constraint in ['fixed', 'bounds']:
+            if constraint in node:
+                param_and_model_constraints[constraint] = node[constraint]
+        model._initialize_constraints(param_and_model_constraints)
+
+        yield model
+
+        if 'inverse' in node:
+            model.inverse = node['inverse']
+
+    def from_yaml_tree(self, node):
+        raise NotImplementedError(
+            "Must be implemented in TransformType subclasses")
+
+    def from_tree(self, node):
+        model = self.from_yaml_tree(node)
+        return self._from_tree_base_transform_members(model, node)
+
+    def _to_tree_base_transform_members(self, model, node):
+        if getattr(model, '_user_inverse', None) is not None:
+            node['inverse'] = model._user_inverse
+
+        if model.name is not None:
+            node['name'] = model.name
+
+        try:
+            bb = model.bounding_box
+        except NotImplementedError:
+            bb = None
+
+        if bb is not None:
+            if model.n_inputs == 1:
+                bb = list(bb)
+            else:
+                bb = [list(item) for item in model.bounding_box]
+            node['bounding_box'] = bb
+        if type(model.__class__.inputs) != property:
+            node['inputs'] = model.inputs
+            node['outputs'] = model.outputs
+
+        # model / parameter constraints
+        if not isinstance(model, CompoundModel):
+            fixed_nondefaults = {k: f for k, f in model.fixed.items() if f}
+            if fixed_nondefaults:
+                node['fixed'] = fixed_nondefaults
+            bounds_nondefaults = {k: b for k, b in model.bounds.items() if any(b)}
+            if bounds_nondefaults:
+                node['bounds'] = bounds_nondefaults
+
+        return node
+
+    def to_yaml_tree(self, model):
+        raise NotImplementedError("Must be implemented in TransformType subclasses")
+
+    def to_tree(self, model):
+        node = self.to_tree_transform(model)
+        return self._to_tree_base_transform_members(model, node)
+
+    def assert_equal(self, a, b):
```

Comment:
These `assert_equal` methods are something that I haven't given much thought yet.  It would be nice to put the responsibility of defining equality onto astropy itself, where upkeep will be easier.

Reply:
Agreed!